server/api/main.py

```diff
@@ -747,25 +747,34 @@ async def _stop_logs():
         await fastapi.concurrency.run_in_threadpool(close_session, db_session)
 
 
-async def _stop_logs_for_runs(runs: list):
-    project_to_run_uids = {}
+async def _stop_logs_for_runs(runs: list, chunk_size: int = 10):
+    project_to_run_uids = collections.defaultdict(list)
     for run in runs:
         project_name = run.get("metadata", {}).get("project", None)
         run_uid = run.get("metadata", {}).get("uid", None)
-        project_to_run_uids.setdefault(project_name, []).append(run_uid)
+        project_to_run_uids[project_name].append(run_uid)
 
     for project_name, run_uids in project_to_run_uids.items():
-        try:
-            await server.api.utils.clients.log_collector.LogCollectorClient().stop_logs(
-                project_name, run_uids
-            )
-        except Exception as exc:
-            logger.warning(
-                "Failed stopping logs for runs. Ignoring",
-                exc=err_to_str(exc),
-                project=project_name,
-                run_uids=run_uids,
-            )
+        if not run_uids:
+            logger.debug("No runs to stop logs for", project=project_name)
```

Comment:
Is this even possible?
You add a project to the dict only if there's a run for that project.

Reply:
just defensive